pulp_rpm/app/tasks/publishing.py

```diff
@@ -542,6 +543,18 @@ def generate_repo_metadata(
     fil_xml.set_num_of_pkgs(total_packages)
     oth_xml.set_num_of_pkgs(total_packages)
 
+    if settings.RPM_METADATA_USE_REPO_PACKAGE_TIME:
+        # gather the times the packages were added to the repo
+        repo_content = (
+            RepositoryContent.objects.filter(
+                repository=publication.repository,
+                version_added__number__lte=publication.repository_version.number,
+            )
+            .exclude(version_removed__number__lte=publication.repository_version.number)
```

Comment:
Strictly speaking, the latest `RepositoryContent` should be the one with a `null` value in the `version_removed` column, and there should only be one of those per content -- right?  So the query could be much simpler?

Reply:
It could be null or it could be a number greater than the repo version being published. So for example if we're publishing version 3, we want to get the `RepositoryContent` records that have version_removed > 3.

Agreed there should be only one RepositoryContent per content. I don't know how I could simplify the query with that though but I'm open to suggestions.

One thing I should mention is that this is copied from [a method on RepositoryVersion in pulpcore](https://github.com/pulp/pulpcore/blob/main/pulpcore/app/models/repository.py#L653). I probably could call that method but I was reluctant to do so since it's marked as private.

Reply:
Ah right, forgot about publishing older versions.  I'll see if this is something we can make public.